common/internal-types/src/channels.rs

```diff
@@ -251,19 +272,21 @@ impl ChannelChange {
     }
 }
 
-/// A wrapper around [`ChannelEntry`] representing a Channel that is corrupted.
+/// A wrapper around [`Hash`] representing a Channel that is corrupted.
 #[derive(Copy, Clone, Debug, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub struct CorruptedChannelEntry(ChannelEntry);
+pub struct CorruptedChannelEntry(Hash);
 
-impl CorruptedChannelEntry {
-    pub fn new(channel: ChannelEntry) -> Self {
-        CorruptedChannelEntry(channel)
+impl From<Hash> for CorruptedChannelEntry {
```

Comment:
Why not keeping `From<ChannelEntry> for CorruptedChannelEntry`? Just passing down the `get_id()`